apps/prairielearn/src/pages/studentDetail/studentDetail.html.tsx

```diff
@@ -0,0 +1,187 @@
+
+interface StudentDetailProps {
+  gradebookRows: StaffGradebookRow[];
+  student: UserDetail;
+  urlPrefix: string;
+  courseInstanceUrl: string;
+}
+
+export function StudentDetail({
+  gradebookRows,
+  student,
+  urlPrefix,
+  courseInstanceUrl,
+}: StudentDetailProps) {
+  const { user, course_instance, enrollment, role } = student;
+
+  const gradebookRowsBySet = new Map<string, StaffGradebookRow[]>();
+  gradebookRows.forEach((row) => {
+    const setHeading = row.assessment_set.heading;
+    if (!gradebookRowsBySet.has(setHeading)) {
+      gradebookRowsBySet.set(setHeading, []);
+    }
+    const setAssessments = gradebookRowsBySet.get(setHeading);
+    if (setAssessments) {
+      setAssessments.push(row);
+    }
+  });
+
+  const handleViewAsStudent = () => {
+    setCookieClient(['pl_requested_uid', 'pl2_requested_uid'], user.uid);
+    setCookieClient(['pl_requested_data_changed', 'pl2_requested_data_changed'], 'true');
+    window.location.href = `${courseInstanceUrl}/assessments`;
+  };
+
+  const handleViewGradebookAsStudent = () => {
+    setCookieClient(['pl_requested_uid', 'pl2_requested_uid'], user.uid);
+    setCookieClient(['pl_requested_data_changed', 'pl2_requested_data_changed'], 'true');
+    window.location.href = `${courseInstanceUrl}/gradebook`;
+  };
+
+  return (
+    <div class="container-fluid">
+      <div class="card mb-4">
+        <div class="card-header bg-primary text-white d-flex align-items-center justify-content-between">
+          <h1 class="mb-0">Details</h1>
+          <button type="button" class="btn btn-sm btn-light" onClick={handleViewAsStudent}>
+            <i class="fas fa-user-graduate me-1" aria-hidden="true" />
+            View as Student
+          </button>
+        </div>
+        <div class="card-body">
+          <h2>{user.name}</h2>
+          <div class="d-flex">
+            <div class="fw-bold me-1">UID:</div>
+            {user.uid}
+          </div>
+          {user.uin && (
+            <div class="d-flex">
+              <div class="fw-bold me-1">UIN:</div> {user.uin}
+            </div>
+          )}
+          <div class="d-flex">
+            <div class="fw-bold me-1">Role:</div> {role}
+          </div>
+          {enrollment?.created_at && (
+            <div class="d-flex">
+              <div class="fw-bold me-1">Enrolled:</div>
+              <FriendlyDate
+                date={enrollment.created_at}
+                timezone={course_instance.display_timezone}
+              />
+            </div>
+          )}
+        </div>
+      </div>
+
+      <div class="card mb-4">
+        <div class="card-header bg-primary text-white d-flex align-items-center justify-content-between">
+          <h2 class="mb-0">Gradebook</h2>
+          <button type="button" class="btn btn-sm btn-light" onClick={handleViewGradebookAsStudent}>
+            <i class="fas fa-book me-1" aria-hidden="true" />
+            View Gradebook as Student
```

Comment:
Suggesting `Sentence casing` here as well. 
```suggestion
            View gradebook as student
```